### PR TITLE
Add `:lang fortran` module

### PR DESCRIPTION
**Pull request by fosskers**
This PR adds support for the [Fortran Language](https://fortran-lang.org/), since an off-the-shelf solution mostly doesn't exist if you aren't using Visual Studio on Windows. Critically, this PR brings together all the treats of modernity recently added to the Fortran world; namely a package manager and LSP. It also bridges the gap between `gfortran` and Intel Fortran, since the latter is much loved but hard to integrate into development environments.

#### TODO

- [x] Basic config skeleton
- [x] `+lsp`
- [x] Common Keybindings
- [x] Compilation / Running support
- ~Optional modeline buttons for building/running (`+buttons`?)~
- ~`+intel`~
- [x] `doom doctor` checks
- [x] README clean up

**Note to Self:** Look into `ob-fortran`.

### CODE REVIEW

File: modules/lang/fortran/config.el [hunk 1]
```diff
@@ -0,0 +1,57 @@
+;;; lang/fortran/config.el -*- lexical-binding: t; -*-
+
+;;
+;;; Packages
+
+(use-package! f90
+  :config
```
Comment (hlissner) [inline modules/lang/fortran/config.el:7]: Add `:defer t` to lazy load the package.

File: modules/lang/fortran/README.org [hunk 2]
```diff
@@ -0,0 +1,76 @@
+#+TITLE:   lang/fortran
+#+DATE:    October 22, 2021
+#+SINCE:   v3.0.0
```
Comment (hlissner) [inline modules/lang/fortran/README.org:3]:
```diff
-#+SINCE:   v3.0.0
+#+SINCE:   v3.0.0 (#5676)
```

**Review by hlissner — CHANGES_REQUESTED** [review]

File: modules/lang/fortran/config.el (hunk 1 above)
Comment (fosskers) [reply modules/lang/fortran/config.el:8]: Done.

File: modules/lang/fortran/README.org (hunk 2 above)
Comment (fosskers) [reply modules/lang/fortran/README.org]: Done.

**Comment by hlissner** [comment]
LGTM! Thanks for your help!

**Comment by fosskers** [comment]
The crossed-off features in the original post will be addressed in follow-up PRs.

**Pull request merged**